BZ1783437 - Add PVC to vSphere

bobfuru opened the pull request:
Adds step to create PVC and reference volumeName as described in [BZ1783437](https://bugzilla.redhat.com/show_bug.cgi?id=1783437).

Preview build to static provisioning steps: https://bz1783437--ocpdocs.netlify.app/openshift-enterprise/latest/storage/persistent_storage/persistent-storage-vsphere.html#vsphere-static-provisioning_persistent-storage-efs

bobfuru commented:
@liangxia or @qinpingli - PTAL

qinpingli commented:
@duanwei33 Please help take a look at this.

duanwei33 commented:
@bobfuru 
1. The volume capacity size is not the same in PV(2Gi) and PVC(5Gi) definition. 
2. For VMDKs creating, I see [vSphere doc](https://vmware.github.io/vsphere-storage-for-kubernetes/documentation/persistent-vols-claims.html#create-vsphere-vmdk) refer to govc and vmkfstools, I'm not familiar with the vmware-vdiskmanager command, how can we get the datastore name after this command, which will be used in PV creating? 
3. For the command "$ vmkfstools -c <size> /vmfs/volumes/DatastoreName/volumes/<disk-name>.vmdk", better to replace "DatastoreName" by "<Datastore-Name>", or we can use the same name according the one PV used.

bobfuru commented:
Thank you, @duanwei33. See my comments below:
 
> 1. The volume capacity size is not the same in PV(2Gi) and PVC(5Gi) definition.
Good catch. I changed capacity to `1Gi` in both places to match another example on the page.

> 2. For VMDKs creating, I see [vSphere doc](https://vmware.github.io/vsphere-storage-for-kubernetes/documentation/persistent-vols-claims.html#create-vsphere-vmdk) refer to govc and vmkfstools, I'm not familiar with the vmware-vdiskmanager command, how can we get the datastore name after this command, which will be used in PV creating?
I'm also not familiar and did not take part in the original creation of this doc. IIUC, the user will only need to include `[datastore-name]` if they used the `vmkfstools` command. I made this more explicit in callout #4.

> 3. For the command "$ vmkfstools -c  /vmfs/volumes/DatastoreName/volumes/.vmdk", better to replace "DatastoreName" by "", or we can use the same name according the one PV used.
I changed it to `<datastore-name>` to match the filename variable at the end of the command.

With these changes, does this look good to you?

duanwei33 commented:
@bobfuru  Thanks for the update, it looks good to me.

lamek commented:
LGTM

bobfuru commented:
/cherrypick enterprise-4.5

bobfuru commented:
/cherrypick enterprise-4.4

bobfuru commented:
/cherrypick enterprise-4.3

bobfuru commented:
/cherrypick enterprise-4.2
